json_encoder.go

```diff
@@ -141,6 +141,12 @@ func (enc *jsonEncoder) AddUint64(key string, val uint64) {
 	enc.bytes = strconv.AppendUint(enc.bytes, val, 10)
 }
 
+func (enc *jsonEncoder) AddUintptr(key string, val uintptr) {
+	enc.addKey(key)
+	enc.bytes = append(enc.bytes, '0', 'x')
+	enc.bytes = strconv.AppendUint(enc.bytes, uint64(val), 16)
```

Comment:
Unfortunately, this isn't valid JSON - the [section of the spec on Numbers](https://tools.ietf.org/html/rfc7159#section-6) specifically requires that numbers be represented in base 10. Instead of this, let's just delegate to `enc.AddInt64`.